### ISSUE TITLE
Remove "open" from name

### ISSUE THREAD

**Issue by CounterPillow** [ValveSoftware/openvr#154]
Since this has no source code anywhere, it's not open source, therefore, you should remove "open" from the "openvr" name so avoid any confusion.

**Comment by Auruss** [comment]
Its just as open as opengl is. Open does not always mean open source. It's open for any hardware manufacturer to create its own drivers.

**Comment by CounterPillow** [comment]
@Auruss I can't find any open specification in this repository, only broken headers and binaries. This is not "Open as in OpenGL". It's a gross misuse of the term "open".

**Comment by SecondReality** [comment]
@CounterPillow I'm curious: why do you consider OpenGL to be open, but not OpenVR?

On specification: There are some details on the api in the wiki, although it's not comprehensive. Would having more details in the wiki fit the bill?

**Comment by stfairy** [comment]
@SecondReality What is the "Khronos Group" for OpenVR?

**Comment by SecondReality** [comment]
@stfairy I'm not aware of a "Khronos Group" for OpenVR.
Are you saying for it to be open there must be an independent non-profit group that can influence the API?

**Comment by Auruss** [comment]
Why do you think the headers are broken? They work fine for me. The specofication you are looking for is the driver api. It's specifies a set of functions and classes that need to be implemented by hardware manufacturers. Exactly the same for opengl.

Also keep in mind that vr is a new thing, and in my opinion it's good to have a company that takes profit from it to push it at the moment. A public and independent comittee would just slow down everything. In the future in makes sense to have a comittee with oculus, valve, microsoft, ...

**Comment by Auruss** [comment]
And as every project it needs time to grow. We just cant expect a specification with good documentation and a commitee out of no where.

**Comment by CounterPillow** [comment]
@Auruss 
>Why do you think the headers are broken? They work fine for me.

Maybe the unmerged PRs and open issues, of which your "open" overlords have addressed exactly 0, should be a hint.

**Comment by zao** [comment]
As a side note, the ultimate goal of all projects is not FL/OSS. Some modules are better served by having one true implementation that can be relied on, instead of the plethora of forks for private purposes that arise if you release something with a too liberal license. The purpose of OpenVR is to be a stable arbiter between applications, drivers and the platform. The FL/OSS movement does not have an monopoly on the term `open`.

**Comment by CounterPillow** [comment]
@zao 

that makes literally zero sense. Just because source code is available and the specification is developed in an open committee does not mean there will be a plethora of forks that people will rely on. On the contrary, it means people can contribute their fixes and improvements upstream and make sure there is one specification and not tons of vendor specific implementations.

All of you people are fairly quick to draw parallels to OpenGL without actually knowing how Khronos Group operates.

**Comment by zao** [comment]
In a world where everyone is responsible and avoids propagating modified versions of the product into the wild, open source makes sense.

It has been shown in other projects with a plugin API that it's very easy for the core to drift in forks and end users throw shared modules together in prayer that they work.

Responsible open source is nice. Not all users of open source are responsible. I wish that you understand the very real problem of fragmentation that has a high chance of occurring when source is available.

A more productive approach would be to ensure that the customization points of the product are so plentiful and properly documented, that the actual product doesn't really need open sourcing as it's just the stable core you can rely on.

**Comment by CounterPillow** [comment]
>I wish that you understand the very real problem of fragmentation that has a high chance of occurring when source is available.

A problem you completely made up. Things fragment when upstream does not merge useful patches. It's what happened with libjpeg-turbo, and now libjpeg-turbo has become the de-facto standard libjpeg. Things fall into place naturally. For libraries, forks are almost always the result of bad upstream management, such as is happening right now with "Open"VR.

**Comment by SecondReality** [comment]
There is another issue along the same lines: https://github.com/ValveSoftware/openvr/issues/8

**Comment by Auruss** [comment]
Well, maybe it makes sense to push the discussion into a different direction. For me 'open' is just a term in the end, it's really generic and could possibly mean anything.
More importantly we should ask ourselves - "is openvr open enough?"
I just want to trigger some brainstorming, so here are some use cases:

1. I'm a developer of a steam-like platform, can I implement OpenVR in such a way that I don't have any dependencies to Steam/SteamVR?
2. We're a small indie company working on their own HMD - will we be able to implement a driver in such a way game developer don't have to care about the actual driver (So it won't require any changes in the game itself)?
3. We're a small indie company working on their own HMD - will we be able to adjust the chaperone bounds because for whatever reason that might be necessary for our HMD

Keep in mind these are just some quick example I just thought about.

1. Probably not, the library will probably look for the SteamVR folder to find the drivers which does not exist in our example, and I also guess there too many missing interfaces (at least documented)
2. Yes! Check out valves GitHub group there is one driver for Razer Hydras
3. Probably not, the drivers don't implement something like that at the moment, so it is SteamVR that  is having this logic, so this relates to my first example.

One last thing, I'm not affiliated with valve but I know that synchronizing a "Open" project with the community is quite hard, so please don't forget that. They are working on it and at the same time we, the community, is also "working" on it. Without proper communication between these two it will just fail.

**Comment by CounterPillow** [comment]
@Auruss 
>They are working on it and at the same time we, the community, is also "working" on it. Without proper communication between these two it will just fail.

In my opinion, that's the big issue here. It seems more like they just dumped the headers and binaries and forgot about it. I realise that Valve's company structure means they don't always have fixed long-term goals on which a dedicated team of people work, but the current state of things here looks like they're working on it internally and occasionally throwing out a release, but not really looking towards working with the community.

I think ideally the spec should be a git repository, with spec changes incorporated from the community, the implementors and the users of the API. These should then be tagged into releases by the upstream management (Valve, in this case), so all parties can claim support for a specific version, and breaking changes can be gathered up and evaluated before being released. Much like with OpenGL, headers can then be automatically generated from the spec.

**Comment by shazow** [comment]
It's important to keep in mind that there are many styles of open source projects, many of which long predate the Github model described here. Almost all of them operate with a "Benevolent dictator" model, and for good reason. It's very difficult to come to consensus quickly and efficiently when there isn't a clear hierarchy of authority. In this case, Valve is the final authority of this particular project just like Linus is the final authority of the Linux kernel and I'm the final authority of my own open source projects.

One of the most important properties of an "open source" project is that it can be forked. You can fork OpenVR. You can make a OpenerVR which _does_ have the full spec version-controlled in the repo (rather than in a wiki), and does a better job at merging pull requests, and spends tons of personal own time and money providing better support for more platforms out of the box without shipping any opaque binaries. You can start a committee to manage OpenerVR in whatever governing fashion you desire. That's your prerogative.

As a maintainer of many open source projects, I am not doing this out of somekind of _obligation_ but rather as a good-will donation to the community. My consumers have no right to _demand_ more of me. I will manage my projects in whatever way fits best with my life, and if that doesn't work for someone then they better put in the work themselves.

Valve has no _obligation_ to keep this open at all, let alone more open than they have. I am very grateful for the work that everyone is putting into this and I hope this project continues to thrive in an increasingly open nature.

If this were my project, I would ask everyone to please keep suggestions constructive and to be prepared to put in the work that they're asking for others to do. This is what open source is about.

**Comment by CounterPillow** [comment]
@shazow 
>It's important to keep in mind that there are many styles of open source projects, many of which long predate the Github model described here. Almost all of them operate with a "Benevolent dictator" model

Which is not mutually exclusive with what I described. I think the benevolent dictator model works great for open-source projects. The current issue is that there appears to be no governing body at all; it's developed without community input, and most of the tracker/PRs are ignored without a comment.

>One of the most important properties of an "open source" project is that it can be forked.

And forking is usually a thing people don't want to do, because they'd rather contribute to the upstream everyone is already used to.

>You can make a OpenerVR which does have the full spec version-controlled in the repo (rather than in a wiki)

This would then be a separate specification, which makes no sense in this context of an open spec. A spec is useless if nobody implements it, so a forked spec would either need to be adopted widely or could never make any changes to what Valve releases, thus not really changing anything at all.

>You can start a committee to manage OpenerVR in whatever governing fashion you desire. That's your prerogative.

Sure, but the path of least resistance is preferred. For me to create an OpenerVR, I would have to:

1. Rewrite binary Valve-provided libraries
2. Get the same kind of marketing power as Valve to get people to use the spec

All for something which I currently have no financial interest in (unlike Valve), and duplicating work Valve has already done.

Therefore, claiming "just fork it if you don't like it" is not constructive. You know as well as everyone else here that neither of us can effectively fork this. In this regard, this is unforkable, since it's a public spec and not an open source software product. And forking is a pain for everyone involved, so it's better for everyone - including upstream - if upstream actually allowed outside contributions.

>As a maintainer of many open source projects, I am not doing this out of somekind of obligation but rather as a good-will donation to the community.

And nothing else was ever implied, so why bring this up?

>and I hope this project continues to thrive in an increasingly open nature.

I hope so too, which is why I want people to discuss this in the first place. Being open in name is pointless if you're not open in nature, and I don't think Valve is doing this out of malice. Having them realise that they're essentially not letting others do free work on their API and surrounding ecosystem for them is an important step towards a more useful project for both Valve and the community.

**Comment by haianos** [comment]
I strongly approve the proposal of removing "open" from the name.
Without entering into details, the term "open" is not generic at all.
This is, at most, "free" software, in the sense that you are free of charge to take it and use it as it is.
The term "open" instead refers to (one of the many!) licenses that protect and rule the sourcecode, in one way or another, the terms of usage, sharing, modification, responsibilities and so on.

The forking problem is indeed not the problem. Firstly, in an opensource license you can explicity mention the rules for forking. Secondly, nothing prevent to Valve to decide exactly which merges to apply to the master branch. Thirdly..that is what github is useful for - sharing code. If you want to share a binary, just a regular download would be more than enough.

Thus, what's the point of having a github repository for something that you don't want to share?

I hope this is going to change soon...and fast....

**Comment by Auruss** [comment]
> Without entering into details, the term "open" is not generic at all.

Of course it is.. the word open has been there long before software development and has been used in many many ways. 

And again, why are we even talking about removing the open in the name instead of talking what valve can do to improve it. They're actively working on it and of course that's difficult. And also they are reading all issues and considering them, I'm pretty sure of that. They already solved a small thing that I wished for in an Issue.

I don't want to repeat myself but please understand that VR is such a early technique, it doesn't make any sense to expect that a perfect open standard comes out of no where.

**Comment by slowriot** [comment]
>I don't want to repeat myself but please understand that VR is such a early technique, it doesn't make any sense to expect that a perfect open standard comes out of no where.

Oculus had an open sourced SDK... how many years ago now?  It's not THAT new an idea, and the standard has no bearing on whether the SDK is open-sourced.  That company already proved it can work fine, having accepted a lot of developer contributions to fatten up their project (before they decided that they no longer support Linux... but that's a different story).

None of that argument applies to the staunch refusal of the managers of this repo to accept pull requests, even ones that are extremely trivial.  For instance, it doesn't take a "benevolent dictator" to sit for hours and decide if [this trivial change](https://github.com/ValveSoftware/openvr/pull/58/files) should be merged or not - it's little more than a typo fix, reluctance to fix which affects real end-user functionality.  The fact that incredibly easy to fix things like these have been ignored speaks volumes of the intentions of those behind OpenVR.  **It's going nowhere**.

**Comment by haianos** [comment]
>>Without entering into details, the term "open" is not generic at all.

>Of course it is.. the word open has been there long before software development and has been used in many many ways.
Indeed. I re-phrase my sentence: in the context of sofware development, the term "open" is not generic at all, but assumes specific, context-dependent annotations.
In this context, it is proposed an implementation shared in a binary form on github platform: the terms "open" here applies to the implementation, which is not open. If the claim is about "open standard", well..such discussion should be public somewhere, and (likely) not on the github platform, which is meant to share code...
That's what the context suggests, at least to me...

>And again, why are we even talking about removing the open in the name instead of talking what valve can do to improve it.
I can't agree more! And I will be extremely happy to do that, but I simply can't without knowing what's behind the screens.

> I don't want to repeat myself but please understand that VR is such a early technique, it doesn't make any sense to expect that a perfect open standard comes out of no where.
It is not - see artoolkit as an example (at least, around since 2001). VR is not early technique, but the acceptance of the technology, the availability of hardware and the commercial opportunities are pushing VR out.

**Comment by sthalik** [comment]
You're piggybacking on open-source nomenclature. Shame.

**Comment by Conzar** [comment]
OpenVR is NOT open source.  The name is intentionally misleading.  Meaning, when users and developers alike see the words 'OpenVR', they immediately think 'Open Source'.  If Volvo doesn't want people to think 'Open Source', than they should change the name to something else.

**Comment by singlerider** [comment]
I agree with @Conzar , except where he mentions Volvo twice. `open` clearly refers to "Open Source", which this is not.

**Comment by Conzar** [comment]
@singlerider volvo is a meme from Dota 2.  I use it here to draw the reader to how Valve has misused the word 'open' by me misusing the word Volvo when its intended that I mean Valve.

**Comment by elect86** [comment]
"Funny" thing: Khronos had to go [OpenXR](https://www.khronos.org/blog/the-openxr-working-group-is-here) because of this...

**Comment by KarjamP** [comment]
Let me just point out that this entire argument assumes the project to consist entirely of the example implementation provided by Valve. It does not. It refers to the API itself, which really is open. This repositary just hosts the library describing the API so that applications can have an easier time implementing OpenVR support within itself.

The library, itself, describing the standard happens to be released under the BSD 3-clause "New" or "Revised" license. This, by sheer definition, makes it both free software and open-source, even if only via the technicality. The only non-free part of the standard happens to be the drivers for OpenVR-compatable devices, with which example implementation provided through Steam happens to be an example of.

You guys are arguing under the assumption that it's the drivers, itself, that make up the entire project, and thus, arguing the name changed due to the drivers not being FOSS software. However, this is not really the case. It's called "OpenVR" because it's meant to be an open standard, one which anyone can use. The GPL has a clause which allows closed-source devices or system applications to be used alongside GPL'd code so long as it provides the necessary functions for applications' use. Since what the application's really doing is talking to the VR devices through the use of a BSD-licensed API, I'd honestly say this counts.

**Comment by sthalik** [comment]
@KarjamP where's the BSD 3-clause spec?

Your line of reasoning regarding GNU GPL wouldn't be accepted in any court. See that even OpenSSL is shaky territory with regard to the GPL.

As for talking to the hardware using BSD-licensed code -- linking to the provided binary-only library 
into its own address space doesn't make it BSD-licensed. GPL is "viral" in that way.

Also, even making a subprocess for the code to circumvent GNU GPL doesn't actually circumvent it. If the subprocess is integral to the software's functioning (there is no replacement), it still falls under the GNU GPL.

Finally, this is as much about Valve piggybacking on open source naming for its benefit, as GNU GPL. I also can't imagine a GNU GPL game or other VR software without linking exceptions. My own software links to a plethora of closed-source SDKs out of necessity of supporting them at all. Yet it has "open" as a prefix because the source code is actually public.

**Comment by TheDeveloperGuy** [comment]
Bla bla bla, what, Valve ate your homework? Who cares what they call it? There's no obligation to be opensource or open anything just because you use the word "open" in your product name. Open for business, open for you to add your headset, open like the gaping maw of Hades, take your pick and find more important things to fight for/about... :P

BTW there is OSVR (Open Source VR) which may satisfy your needs for a VR operating environment to be (mostly) open source. No offense, I'm going to assume Valve's developers are better than you anyway when it comes to developing VR operating environments, so apart from satisfying some "all software must be free" naivety, what do you want from the source code anyway? And if you don't want the source code, why do you care at all about whether it is opensource or not?

**Comment by KarjamP** [comment]
In reply to sthalik:

First off, let me point out that the Free Software Federation does not consider "Open Source" to be the same thing as "Free Software". See [this](https://www.gnu.org/philosophy/open-source-misses-the-point.en.html) for more info.

Second, if the problem really is that it can't be BSD'd due to the fact that only a binary seems to be provided, that's an altogether different issue than this one. After all, the license to the API claims it to be under that license. If it can't due to that one simple technicality...

Third, if your line of reasoning regarding driver code not being compatable with GPL'd software holds sound, that means none of the software released for Windows would be legal (since most Windows drivers happen to be closed source), and you can't legally do things like install propietary Nvidia drivers on your Linux-based computer. The license has that concensus for a reason.

Fourth, you can't compare OpenVR to OpenSSL since OpenSSL used a decisively non-compatible license, one entirelly different from the one OpenVR's supposed to have. Also, you can't just speak for the court; stuff like this are usually determined on a case-by-case context due to the sheer complexity of the law itself.

**Comment by sthalik** [comment]
@KarjamP,

> Also, you can't just speak for the court; stuff like this are usually determined on a case-by-case context due to the sheer complexity of the law itself.

GNU GPL requires all of the code to be released, even including the full build system.

From <<https://www.gnu.org/licenses/quick-guide-gplv3.en.html>>

> Less Source to Distribute: New System Libraries Exception
Both versions of the GPL require you to provide all the source necessary to build the software, including supporting libraries, compilation scripts, and so on.  [...]

Look it up some more if you don't believe me. It's established knowledge. I wouldn't say of it as a certainty otherwise.

This GitHub issue really is something else. I shouldn't be participating here.

**Comment by KarjamP** [comment]
As I said, that's an altogether different issue than the one presented here. It doesn't matter whether or not the project, as presented here, is GPL compatable. The actual argument is that the word "open" should be dropped from the name since it's "misleading". It's possible for something to be open without it being GPL-compatable.

Since you already know this, why are you bringing it up? To do so means to know you're diverging from the main topic at hand.

**Comment by sthalik** [comment]
@KarjamP You went on a tangent about GNU GPL compatibility. I'm deleting my posts here, it reflects badly on me to participate in this pissing match.

**Comment by Conzar** [comment]
> OpenVR refers to the standard itself, which really is open. This repository just provides an API allowing applications to use OpenVR. - KarjamP

According to Wikipedia:
> OpenVR is a software development kit and application programming interface developed by Valve for supporting the SteamVR (HTC Vive)[1][2] and other virtual reality headset devices.
 - [link](https://en.wikipedia.org/wiki/OpenVR)

Since the software development kit is not OPEN, by the very definition of OpenVR provided by Wiipedia, OpenVR is not OPEN.

**Comment by KarjamP** [comment]
In case anyone's wondering,  sthalic deleted his posts, apparently because the discussion here reflected badly on him (his words). Therefore, I was not simply talking to myself, here.

By the way, this project really is GPL-compatable. Sthalic was arguing that it wasn't due to the source code not being distributed. However, he seems to be confusing SteamVR, a "driver" thst supports OpenVR, with OpenVR itself. In fact, the source code to OpenVR, or at least the API, itself, really is distributed to the public. It's within the "src" directory of the project. This is what gets compiled when you tell Cmake to do so.

To be honest, how OpenVR works is no different than how OpenGL, itself works. Fun fact: On many ststems, OpenGL also requires applications to load a dynamic library. It's just not that obvious since the API's for the standard itself happens to be included within many compilers.

Also, as an additional note, Wikipedia shouldn't be a deciding factor as to whether or not something happens to be so. Yes, Wikipedia aims to be a reliable sorce, but the fact still remains that anyone can say anything, there. If you are to use Wikipedia as a source, use whatever they link as reference instead.

**Comment by Conzar** [comment]
> but the fact still remains that anyone can say anything, there. If you are to use Wikipedia as a source, use whatever they link as reference instead. - KarjamP

You are some random person on the internet who is saying 'anything'.  Where are your sources for YOUR definition of OpenVR?  Because its 'YOUR' definition you made up in order to support your invalid argument.

**Comment by KarjamP** [comment]
I never said I was defining something. Now, instead of attacking me, why not just refute my claims?

**Comment by Conzar** [comment]
> I never said I was defining something. - KarjamP

> OpenVR refers to the standard itself, which really is open. - KarjamP

Seems clear enough you have a definition of OpenVR you are using in your argument.  This definition does not comply with the wikipedia definition which is well sourced.  Its up to you to follow the link to the wikipedia article and to follow the 4 references.

> Now, instead of attacking me, why not just refute my claims?
Please provide evidence that I am 'attacking' you.  I have already refuted your argument based on the definition of OpenVR. 

I will ask you politely to apologise for your fallacious claim that I am attacking you.  You have lost the argument and your last recourse is to make fallacious claims.  This behaviour is unacceptable in civil discussion.

**Comment by KarjamP** [comment]
If I think Wikipedia's not a reliable source, what makes you think I'm going to even visit Wikipedia? You have a hostile tone to your words. This is my evidence that you're attacking me.

Ad Hominem isn't the only logical fallacy that you're showing me. You're also showing me "appeal to authority". In addition, even if I were to give you evidence, what's to say you'd be convincee? Why not argue against my own words instead of asking for evidence? After all, the "burden of proof", especially when used wrong, also happens to be a logical fallacy in of itself.

Seriously, attack the argument instead of me. Attack the argument instead of questioning the sources of my information, which is also an attack against me. I only did so because to assume Wikipedia as an absolute is a form of  "appealing to authority". Even the most seemingly reliable of sources can be wrong at times; I only suggested to use the sources it provides instead because the nature of Wikipedia as an encyclopedia anyone can edit makes it a lot more riskier as a mean of gathering information than the sources it provides. Finally, attack the argument instead of asking for information.

Seriously, if you're not going to properly argue for this issue, why are you even doing so at all? We're supposed to be discussing why OpenVR's not open, not questioning our sources of information or asking for evidence.

**Comment by Conzar** [comment]
@KarjamP I have presented my case and have substantiated it.  You have made your case and have not substantiated it.  Now you have slandered me without providing ANY evidence.  I am blocking you as you clearly do not understand how to properly communicate.

**Comment by echuber2** [comment]
The SteamVR runtime is not open-source, but the top post here is barking up the wrong tree. The question is whether it's an open _standard_, not open _source_. OpenGL isn't open source either; it has no source code, because it's a standard, not an implementation. Refer to:

> Is OpenGL Open Source?
> 
> No, OpenGL doesn't have any source code. GL is a specification which can be found on this website. It describes the interface the programmer uses and expected behavior. OpenGL is an open specification. Anyone can download the spec for free. This is as opposed to ISO standards and specifications, which cost money to access.
> 
> There is an implementation of GL that is Open Source and it is called Mesa3D. The latest versions implement OpenGL 4.3, OpenGL ES 2.0, GLSL 3.30 (and several extensions) and EGL 1.4.

from https://www.khronos.org/opengl/wiki/FAQ#Is_OpenGL_Open_Source.3F

And this quote from Khronos should silence most of the debate above. GPL has nothing to do with this since OpenVR isn't under GPL.

OpenVR is merely a set of header files you can download for free. You can (realistically) implement your own drivers for your own HMD or (unrealistically) your own runtime for someone else's HMD. If you want your games to run on Steam, they have to use the closed-source SteamVR implementation.

As to the latter part, I do wish Valve would publish more details about how exactly the SteamVR implementation works in some aspects. Their GDC talks reveal some useful details, as do the bread crumbs in the docs here and on Steam forums, and the devs have been answering questions on here some of the time. Mr. Leiby was quite helpful in me gaining a better understanding of how the pose prediction works in fine detail. But of course it might have been simpler if I could have just taken a look at their code. Occasionally the Valve employees have pasted a few key lines of the closed source for SteamVR on the forums to clear up some detail.

**Comment by KarjamP** [comment]
It is an open standard in regards to how you said it. Just because SteamVR, itself, isn't an open sourced implementation does not make OpenVR, itself, even less so. It's just that there's not enough options for more hardware to use with the standard itself.

[Here's an example of a custom driver for a device commonly used for VR](https://github.com/cbuchner1/driver_leap). This proves that it's possible to develop new drivers for the standard. The fact that it's forked from Valve's own open sourced drivers for the relatively ancient device called the "Razer Hydra" also proves the existence of at least one open source driver intended for use as a reference.

(By the way, the OpenGL libraries often included with compilers are oftentimes, also header files, themselves. You're not really compiling to OpenGL anymore than using an API designed to load a library that provides graphcs card functions. Said library are oftentimes, closed source binaries as well, especially on Windows computers.)

**Comment by echuber2** [comment]
OpenGL libraries including the header files are provided with the implementation, so as such, they aren't the thing OpenGL itself. For example on my Linux system the GL.h is just a file provided by Mesa. OpenGL itself is nothing but the very boring technical details of how it's supposed to work, in tables and charts. So what I'm saying is that the question of whether the runtimes or binaries are closed or not is totally orthogonal to the question of whether this is a open standard or not, which is definitely debatable.

Like I mentioned, to begin to be on par with OpenGL, a lot of the details from the one-on-one discussions with Valve employees, the GDC talks, etc. need to be properly distilled into formal documentation and included on this repo, in my opinion. Theoretically, you should be able to develop your own competing runtime, not only the HMD drivers. If the spec isn't complete enough to develop a runtime that is compatible, I think there's a way to go.

**Comment by KarjamP** [comment]
The same logic can be used for OpenVR itself. The reason why there doesn't seem to be open binaries is because those binaries everyone's been mentioning happens to be a mere implementation of OpenVR, but not OpenVR itself. The same goes for the source code provided in this repository. The source code's meant to help applications gain support for this standard, but if your logic holds sound, that source code wouldn't be OpenVR itself.

If you were to tell CMake that you want to build this project, what would you get happens to be a dynamically linked library which allows applications to talk with the OpenVR drivers. This is no different than, to say, building the [Vulkan loader](https://www.lunarg.com/vulkan-sdk/). (Can't use OpenGL as a reference; those headers are often included alongside compilers, and as such, might be more difficult to discover.)

**Comment by Conzar** [comment]
What is OpenVR?  According to the wikipedia article I referenced, its both an API and a software development kit.  According to Valve's githup page:

> OpenVR is an API and runtime that allows access to VR hardware from multiple vendors without requiring that applications have specific knowledge of the hardware they are targeting.
 - [reference](https://github.com/ValveSoftware/openvr)

I think we all can agree, the runtime is proprietary.  The runtime is apart of the definition BY VALVE.  So when looking at Valve's definition that OpenVR is both an API and runtime, its clear that OpenVR is NOT OPEN due to the second clause of their sentence.  "This of course should silence most of the debate above."

We don't need to investigate any further into the API as I have already proven that OpenVR according to valve's definition is NOT OPEN.

But just as an exercise, who has control over the API?  Is it a group, a committee, or is it just Valve? 
Open Standard has a [wide range](https://en.wikipedia.org/wiki/Open_standard) of definitions.  My opinion of the API is that its proprietary and owned by Valve.

**Comment by TheDeveloperGuy** [comment]
Round and round we go. Use of the word "open" does not belong to anyone (ie. it's not a trademark and noone can enforce what it is used for). You may think it has certain implications, and you may be unhappy that it doesn't meet what is required to match your expectations. That doesn't change the fact that Valve can call it whatever they like, and do whatever they want with it. It is theirs. There are not a lot of people who care, nor should there be. ;)

**Comment by Conzar** [comment]
>  That doesn't change the fact that Valve can call it whatever they like, and do whatever they want with it. It is theirs.

What you describe is proprietary.  Its the opposite of open.  Valve can do what it wants and its customers are free to make complaints here describing why we think OpenVR is mislabelled.

**Comment by ChristophHaag** [comment]
>  The runtime is *a part* of the definition BY VALVE.

Indeed, but you can also read it in a way that they mean *any* runtime. It just so happens that nobody else seems to have developed one.

If anyone wants to help with one: https://github.com/ChristophHaag/openvr_api-libre

**Comment by ChristophHaag** [comment]
To be fair, Valve's libopenvr_api in this repository really isn't fully open, for example it contains an undocumented `VRDashboardManager` that is not present when you compile libopenvr_api yourself: https://github.com/ValveSoftware/openvr/issues/425

As far as I'm aware only SteamVR itself (like SteamVR/bin/linux64/vrdashboard ) needs this stuff and all 3rd party OpenVR applications should still run without it.
